### DISCUSSION TITLE
Enabling monitoring port for kafka/zookeeper

### DISCUSSION THREAD

**Discussion by pashafirdous** [strimzi/strimzi-kafka-operator#4271] [Q&A]
Hi,
could you please suggest a method to enable prometheus monitoring port 9404 for kafka/zookeeper?

we understand it is deprecated in 0.19.0 onwards but we use monitoring tool which will listen on a port and scrape metrics.

need help

**Comment by scholzj** [comment]
The port is still present in the pods when the metrics are enabled and can be used. It was sonly removed from the services. If you really need it on the service, you can create your own service with the port and custom name manually. But if you for example use Prometheus, you should be able to just configure it to get the data from pods directly (the service is anyway used only to discover the pods).

**Reply by saloni-2404** [reply]
Hey Jakub,
I tried to create a service for kafka like the following:

apiVersion: v1
kind: Service
metadata:
  labels:
    app.kubernetes.io/instance: my-cluster-1
    app.kubernetes.io/managed-by: strimzi-cluster-operator
    app.kubernetes.io/name: kafka
    app.kubernetes.io/part-of: strimzi-my-cluster-1
    strimzi.io/cluster: my-cluster-1
    strimzi.io/kind: Kafka
    strimzi.io/name: my-cluster-1-kafka
  name: my-cluster-1-kafka-aiops
  namespace: kafka-ver-20-sal
spec:
  ports:
  - name: tcp-external
    port: 9404
    protocol: TCP
    targetPort: 9404
  selector:
    strimzi.io/cluster: my-cluster-1
    strimzi.io/kind: Kafka
    strimzi.io/name: my-cluster-1-kafka
  sessionAffinity: None
  type: NodePort

This had worked for zookeeper, kafkaexporter. But for kafka, it comes up, gets automatically removed. What am i doing wrong here?

**Reply by scholzj** [reply]
Yeah ... you should not copy all the Strimzi labels in the service metadata. That basically says that the operator owns it and it might delete it. Just remove the `strimzi.io/...` labels.

**Reply by saloni-2404** [reply]
That seems to help, its did not get deleted so far, thank you Jakub!
